test/test_nn.py

```diff
@@ -10431,6 +10418,50 @@ def test_multi_margin_loss_errors(self, device):
                           lambda: nn.functional.multi_margin_loss(torch.randn(5, device=device),
                                                                   torch.zeros(3, device=device)))
 
+    def test_nll_loss_mismatched_batch(self, device):
+        x = torch.randn((10, 3), requires_grad=True, device=device)
+        # t should have size (10,)
+        t = torch.zeros((3,), dtype=torch.int64, device=device)
+        with self.assertRaisesRegex(ValueError, 'Expected.*batch_size'):
+            F.nll_loss(x, t)
+
+    def test_nll_loss_out_of_bounds_ignore_index(self, device):
+        x = torch.randn(6, 3, requires_grad=True, device=device)
+        t = torch.tensor([0, 1, 255, 0, 1, 2], dtype=torch.int64, device=device)
+        for reduction in ['mean', 'none']:
+            F.nll_loss(x, t, ignore_index=255, reduction=reduction).sum().backward()
+
+    def test_nll_loss_empty_tensor(self, device):
+
+        def helper(input_size, reduction, expected):
+            input = torch.rand(input_size, requires_grad=True, device=device)
+            num_channels = input_size[1]
+            target_size = (input_size[0], ) + tuple(input_size[2:])
+            target = torch.randint(num_channels, target_size, device=device)
+
+            output = F.nll_loss(input, target, reduction=reduction)
+            self.assertEqual(output, expected)
+
+            output.sum().backward()
+            self.assertEqual(input.grad.size(), input.size())
+
+        helper([0, 3], "none", torch.empty([0], device=device))
+        helper([0, 3, 5, 7], "none", torch.empty([0, 5, 7], device=device))
+        helper([2, 3, 0, 7], "none", torch.empty([2, 0, 7], device=device))
+        helper([2, 3, 5, 0], "none", torch.empty([2, 5, 0], device=device))
+
+        nan = torch.tensor([float('nan')], device=device).resize_([])
```

Comment:
Correct me if I'm wrong, but `torch.tensor([float('nan')]) = tensor([nan])` and size is [1]. If I resize it then it became `tensor(nan)` with size []. I expect size to be [], not [1].

Reply:
Everything you said is correct. I was suggesting that instead of `torch.tensor([float('nan')], device=device).resize_([])`, we use `torch.tensor(float('nan'), device=device)`, because those two expressions are equivalent (they both have size []). For example:
```
import torch
nan1 = torch.tensor([float('nan')]).resize_([])
nan2 = torch.tensor(float('nan'))

>>> print(nan1.size())
torch.Size([])

>>> print(nan2.size())
torch.Size([])
```

Reply:
thanks! addressed